controllers/volumereplication_controller.go

```diff
@@ -42,17 +45,43 @@ type VolumeReplicationReconciler struct {
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the VolumeReplication object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
 //
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.7.0/pkg/reconcile
 func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = r.Log.WithValues("volumereplication", req.NamespacedName)
 
-	// your logic here
+	_ = r.Log.WithValues("controller_volumereplication", req.NamespacedName)
+
+	// Fetch VolumeReplication instance
+	instance := &replicationv1alpha1.VolumeReplication{}
+	err := r.Client.Get(context.TODO(), req.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			r.Log.Info("no VolumeReplication resource found")
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	var volumeHandle string
+	dataSourceName := instance.Spec.DataSource.Name
+	switch instance.Spec.DataSource.Kind {
+	case "PersistentVolumeClaim":
+		_, pv, err := r.getPVCDataSource(req.NamespacedName, dataSourceName)
```

Comment:
Why not create `NamespacedName` with PVC name and namespace? That's 1 less param.